src/app/playground/[type]/page.tsx

```diff
@@ -0,0 +1,89 @@
+"use client";
+import { useEditorTheme } from "@/hooks/useEditorTheme";
+import { selectTheme } from "@/redux/features/theme/themeSlice";
+import { useAppSelector } from "@/redux/hooks";
+import { LanguageType } from "@/utils/types/languages";
+import { Editor } from "@monaco-editor/react";
+import { useState } from "react";
+import axios from "axios";
+import startingCodeMap from "@/utils/playground/starting-code";
+
+type Props = {
+  params: {
+    type: LanguageType;
+  };
+};
+
+const Playground = ({ params: { type } }: Props) => {
+  const setEditorTheme = useEditorTheme();
+  const theme = useAppSelector(selectTheme);
+  const [userCode, setUserCode] = useState(startingCodeMap[type]);
+  const [consoleLog, setConsoleLog] = useState("");
+  const apiUrl = process.env.NEXT_PUBLIC_RUNCODE_API || "";
+
+  function handleCodeChange(value: string | undefined) {
+    if (value) {
+      setUserCode(value);
+    }
+  }
+
+  async function runCode() {
+    try {
+      const result = await axios.post(apiUrl, {
+        code: userCode,
+        language: type,
+      });
+      setConsoleLog(result.data?.log || "");
+    } catch (error) {
+      console.log("error:", error);
+    }
+  }
+
+  return (
+    <div className="h-[calc(100vh-64px)] border md:mx-20 lg:mx-32 xl:mx-48">
```

Comment:
1. Maybe add a secondary navigation bar or dropdown menu to let user navigate between different languages more easily.
1. I think using a state to keep track of the language type is enough. Re-render the editor with corresponding language. This can reduce the number of GET requests.
1. Display a tag or some sort to make the user aware of which language's the playground is using.

Reply:
Let's do this in the next PR, we need to create a Editor component to be used in both problem page and playground page.